src/Proto3/Suite/Form/Encode.hs

```diff
@@ -408,7 +408,10 @@ instance ProtoEnum e =>
 
 instance FieldForm 'Optional 'Bytes (Identity RB.BuildR)
   where
-    fieldForm _ _ !fn (Identity x) = Encode.bytes fn x
+    fieldForm _ _ = coerce
+      @(FieldNumber -> RB.BuildR -> Encode.MessageBuilder)
+      @(FieldNumber -> Identity RB.BuildR -> Encode.MessageBuilder)
```

Comment:
I don't know if you've ever used this before, but you can use wild cards in type applications and GHC will fill the hole with inference. So you could shorten this to `@(_ -> _ RB.BuildR -> _)` I think, if you wanted.